pkg/mover/rewrite_rules.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+package mover
+
+import (
+	"errors"
+
+	"github.com/docker/distribution/reference"
+)
+
+// RewriteRules indicate What kind of target registry overrides we want to apply to the found images
+type RewriteRules struct {
+	// Registry overrides the registry part of the image FQDN, i.e myregistry.io
+	Registry string
+	// RepositoryPrefix will override the image path by being prepended before the image name
+	RepositoryPrefix string
+}
+
+func (r *RewriteRules) Validate() error {
+	if r.Registry != "" {
+		reference.DomainRegexp.Longest()
+		matched := reference.DomainRegexp.FindString(r.Registry) == r.Registry
+		if !matched {
+			return errors.New("registry rule is not a valid registry domain")
```

Comment:
to be consistent with the error message below, maybe we should just say "registry is not valid"?

Reply:
Sounds good. I'll adjust that.